src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -633,7 +633,19 @@ public virtual SourceText WithChanges(IEnumerable<TextChange> changes)
                 // there can be no overlapping changes
                 if (change.Span.Start < position)
                 {
-                    throw new ArgumentException(CodeAnalysisResources.ChangesMustBeOrderedAndNotOverlapping, nameof(changes));
+                    // Handle the case of unordered changes by sorting the input and retrying. This is inefficient, but
+                    // downstream consumers have been known to hit this case in the past and we want to avoid crashes.
+                    // https://github.com/dotnet/roslyn/pull/26339
+                    if (change.Span.End <= changeRanges.Last().Span.Start)
+                    {
+                        changes = from c in changes
+                                  where !c.Span.IsEmpty || c.NewText.Length > 0
+                                  orderby c.Span.Start, c.Span.End
+                                  select c;
+                        return WithChanges(changes);
```

Comment:
>WithChanges [](start = 31, length = 11)

Do we have tests that demonstrate the new behavior?

Reply:
> Do we have tests that demonstrate the new behavior?

No. Local install was broken due to NRT preview install so I had to fix that, and now am working on tests. Hence the **PR for Personal Review Only** label. 😄

Reply:
➡️ Tests are added